distribution/tools/plugin-cli/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -598,25 +598,28 @@ private PluginInfo verify(Terminal terminal, Path pluginRoot, boolean isBatch, E
 
     /** check a candidate plugin for jar hell before installing it */
     void jarHellCheck(PluginInfo candidateInfo, Path candidateDir, Path pluginsDir, Path modulesDir) throws Exception {
-        // create list of current jars in classpath
-        final Set<URL> jars = new HashSet<>(JarHell.parseClassPath());
-
-
-        // read existing bundles. this does some checks on the installation too.
-        Set<PluginsService.Bundle> bundles = new HashSet<>(PluginsService.getPluginBundles(pluginsDir));
-        bundles.addAll(PluginsService.getModuleBundles(modulesDir));
-        bundles.add(new PluginsService.Bundle(candidateInfo, candidateDir));
-        List<PluginsService.Bundle> sortedBundles = PluginsService.sortBundles(bundles);
-
-        // check jarhell of all plugins so we know this plugin and anything depending on it are ok together
-        // TODO: optimize to skip any bundles not connected to the candidate plugin?
-        Map<String, Set<URL>> transitiveUrls = new HashMap<>();
-        for (PluginsService.Bundle bundle : sortedBundles) {
-            PluginsService.checkBundleJarHell(bundle, transitiveUrls);
-        }
+        try {
+            // create list of current jars in classpath
+            final Set<URL> jars = new HashSet<>(JarHell.parseClassPath());
+            // read existing bundles. this does some checks on the installation too.
+            Set<PluginsService.Bundle> bundles = new HashSet<>(PluginsService.getPluginBundles(pluginsDir));
+            bundles.addAll(PluginsService.getModuleBundles(modulesDir));
+            bundles.add(new PluginsService.Bundle(candidateInfo, candidateDir));
+            List<PluginsService.Bundle> sortedBundles = PluginsService.sortBundles(bundles);
+
+            // check jarhell of all plugins so we know this plugin and anything depending on it are ok together
+            // TODO: optimize to skip any bundles not connected to the candidate plugin?
+            Map<String, Set<URL>> transitiveUrls = new HashMap<>();
+            for (PluginsService.Bundle bundle : sortedBundles) {
+                PluginsService.checkBundleJarHell(bundle, transitiveUrls);
+            }
+            // TODO: no jars should be an error
+            // TODO: verify the classname exists in one of the jars!
+        } catch (IllegalStateException e) {
```

Comment:
I do not think this is right, we could catch an IllegalStateException for JAR hell and really we are only trying to detect when the offline plugin is contained in the plugins folder.